src/components/searchInput/searchInput.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { Person } from '../../types/Person';
+
+interface Props {
+  delay: number;
+  setFullField: React.Dispatch<React.SetStateAction<string>>;
+  setSelectedPerson: React.Dispatch<React.SetStateAction<null | Person>>;
+  selectedPerson: null | Person;
+  setFocus: React.Dispatch<React.SetStateAction<boolean>>;
+  setDelay?: React.Dispatch<React.SetStateAction<number>>;
+}
+
+export const SearchInput: React.FC<Props> = ({
+  delay,
+  setFullField,
+  selectedPerson,
+  setSelectedPerson,
+  setFocus,
+}) => {
+  const [field, setField] = useState('');
+
+  const timerId = useRef<NodeJS.Timeout | number>(0);
+
+  const previousField = useRef('');
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setSelectedPerson(null);
+    setField(e.target.value);
+
+    if (timerId.current) {
+      window.clearTimeout(timerId.current as number);
+    }
+
+    timerId.current = window.setTimeout(() => {
+      if (previousField.current !== e.target.value) {
+        const newField = e.target.value;
+
+        setFullField(newField.trim());
```

Comment:
The `setFullField` function is correctly trimming the input value. However, ensure that the logic in the parent component handles cases where the trimmed value is an empty string, as this is a core requirement.